internal/policy/verify_test.go

```diff
@@ -728,65 +728,222 @@ func TestVerifyRelativeForRef(t *testing.T) {
 		err = VerifyRelativeForRef(context.Background(), repo, policyEntry, nil, policyEntry, entry, refName)
 		assert.ErrorIs(t, err, ErrUnauthorizedSignature)
 	})
-}
 
-func TestVerifyCommit(t *testing.T) {
-	repo, _ := createTestRepository(t, createTestStateWithPolicy)
-	refName := "refs/heads/main"
-	gpgKey, err := gpg.LoadGPGKeyFromBytes(gpgPubKeyBytes)
-	if err != nil {
-		t.Fatal(err)
-	}
+	t.Run("no recovery, with attestations", func(t *testing.T) {
```

Comment:
I haven't looked through the tests yet, I think it can wait until you've had a chance to look through my comments above.